src/groth16/aggregate/accumulator.rs

```diff
@@ -71,15 +79,30 @@ impl<E: Engine, R: rand::RngCore + Send> PairingChecks<E, R> {
         self.merge(PairingCheck::from_miller_one(result), must_randomize);
     }
 
-    pub fn merge_pairing_equation(&self, left: Vec<E::Fqk>, pair: (E::Fqk, E::Fqk)) {
-        let must_randomize = self.non_random_check_done.load(SeqCst);
-
-        for l in left.iter() {
-            self.merge_miller_one(*l, must_randomize);
+    /// takes a vector of elements  which are outputs of miller outputs and an
+    /// right element which fulfills the following:
+    ///
+    /// $$
+    /// \prod left_i = right
+    /// $$
+    ///
+    /// It can only be called ONCE as this is the only non random check allowed.
+    /// It panics if called more than once.
+    pub fn merge_nonrandom(&self, left: Vec<E::Fqk>, right: E::Fqk) {
+        let randomize = self.non_random_check_done.load(SeqCst);
+        self.merge_pair(left[0], right, randomize);
```

Comment:
This will fail if called on an empty `left`. To be even more general, we would handle that case also, but this is fine for present purposes.